core/platform/lf_zephyr_support.c

```diff
@@ -367,7 +374,8 @@ int lf_available_cores() {
 int lf_thread_create(lf_thread_t* thread, void *(*lf_thread) (void *), void* arguments) {
     // Use static id to map each created thread to a 
     static int tid = 0;
-    
+
+    // Make sure we dont try to create too many threads
```

Comment:
If `lf_thread_create` is called in user code, then the number of workers isn't really relevant, is it? Does this really belong here?

Reply:
I think it is bad practice to do too much dynamic spawning of threads in a microcontroller setting. You want to avoid dynamic memory allocation if you can. Therefore the stacks and thread control blocks are statically allocated. This puts an upper limit to how many times you can call lf_thread_t. The user can use ordinary calls to zephyrs API to create other threads.

Reply:
While I don't disagree with you on this, AFAIK, the Zephyr implementation is the only one that does this. My point of criticism has to do with the fact that the function is `lf_thread_create` not `lf_worker_create`. Threads and workers are two different things, and you're now looking at the number of workers in the thread creation function, which I think is odd...

Reply:
Hm, I would argue that reactor-c is not a threading library and that user-code should not use the LF threading API. That would make the runtime more attractive for use in these embedded systems where people are wary of using dynamic memory allocation. There is already a lot of that in the runtime though, but theoretically, most of it can be replaced by static allocations.

Reply:
The function does not have a leading underscore, so it is part of the public API...

Reply:
Zephyr has a POSIX implementation that we could use temporarily, or we can just turn of threaded support for Zephyr, that is fine with me. With the bodiless reactions we have to make a decision about what is actually going to be part of the user-facing API since those includes must be code-generated. So let's make the right decision there and revisit this later

Reply:
I'm not sure I understand your proposal. Here are some possible solutions that crossed my mind:
 - have a define `MAX_THREAD_COUNT` and check against that
 - update the docs and have all implementations check against `NUMBER_OF_WORKERS`
 - rename to `_lf_thread_create`
 - introduce a `lf_worker_create` that does the check and then just invokes `lf_thread_create`
 - ...?